dspace-api/src/main/java/org/dspace/storage/bitstore/AzureBlobStorageBitStoreService.java

```diff
@@ -0,0 +1,395 @@
+			}
+
+			this.blobContainer = this.serviceClient.getContainerReference(getAzureContainerReference());
+
+			if (!this.blobContainer.exists()) {
+				this.blobContainer.create();
+				log.info("Creating new Azure Container: ".concat(getAzureContainerReference()));
+			}
+		} catch (Exception e) {
+			log.error(e);
+			throw new IOException(e);
+		}
+
+		log.info("Azure Blob Storage is ready to go! Container Reference: ".concat(getAzureContainerReference())
+				.concat(System.getProperty("line.separator")));
+	}
+
+	/**
+	 * Returns an identifier unique to this asset store instance
+	 * 
+	 * @return a unique ID
+	 */
+	@Override
+	public String generateId() {
+		return Utils.generateKey();
+	}
+
+	/**
+	 * Retrieve the bits for the asset with ID. If the asset does not exist, returns
+	 * null.
+	 * 
+	 * @param bitstream
+	 *            The ID of the asset to retrieve
+	 * @exception java.io.IOException
+	 *                If a problem occurs while retrieving the bits
+	 *
+	 * @return The stream of bits, or null
+	 */
+	@Override
+	public InputStream get(Bitstream bitstream) throws IOException {
+		String key = getFullKey(bitstream.getInternalId());
+		try {
+			CloudBlob blob = this.blobContainer.getBlockBlobReference(key);
+			return (blob != null) ? blob.openInputStream() : null;
+		} catch (Exception e) {
+			log.error("get(" + key + ")", e);
+			throw new IOException(e);
+		}
+	}
+
+	/**
+	 * Store a stream of bits.
+	 *
+	 * <p>
+	 * If this method returns successfully, the bits have been stored. If an
+	 * exception is thrown, the bits have not been stored.
+	 * </p>
+	 *
+	 * @param in
+	 *            The stream of bits to store
+	 * @exception java.io.IOException
+	 *                If a problem occurs while storing the bits
+	 */
+	@Override
+	public void put(Bitstream bitstream, InputStream inputStream) throws IOException {
+		String key = getFullKey(bitstream.getInternalId());
+		// Copy istream to temp file, and send the file, with some metadata
+		File scratchFile = File.createTempFile(bitstream.getInternalId(), "azureblobstorage");
+		try {
+			FileUtils.copyInputStreamToFile(inputStream, scratchFile);
+			Long contentLength = Long.valueOf(scratchFile.length());
+
+			CloudBlockBlob blob = this.blobContainer.getBlockBlobReference(key);
+
+			this.uploadFileBlocksAsBlockBlob(blob, scratchFile.getAbsolutePath());
+
+			bitstream.setSizeBytes(contentLength);
+			bitstream.setChecksum(blob.getProperties().getEtag());
+			bitstream.setChecksumAlgorithm(this.algorithm);
+
+			scratchFile.delete();
+
+		} catch (Throwable t) {
+			log.error("put(" + bitstream.getInternalId() + ", is)", t);
+
+			StringWriter stringWriter = new StringWriter();
+			PrintWriter printWriter = new PrintWriter(stringWriter);
+			t.printStackTrace(printWriter);
+			if (t instanceof StorageException) {
+				if (((StorageException) t).getExtendedErrorInformation() != null) {
+					log.error(String.format("\nError: %s",
```

Comment:
Instead of this `if`, this should be a separate `catch` statement. Again, the `log.error` should provide more information about why a StorageException error may occur here.